### ISSUE TITLE
[Issue]: cant use openOutpaint, "--api" not available

### ISSUE THREAD

**Issue by GeorgLegato** [vladmandic/automatic#282]
### Issue Description

Using Vlad1111 on Mac M1.
Starting "sh. webui.sh" with COMMAND_ARG=--listen
openOutpaint is "offline". From Automatic1111 I know it uses "--api", but this cmd arg is not present in Vlads version

Please help howto to get openOutpaint up

### Version Platform Description

Mac M1 Max
Osx 12.3
FireFox
latest Vladmatic1111

**Comment by vladmandic** [comment]
no extension should check if command line flag is present, that is just silly. it should check if functionality is available - and in this repo, `api` is always-on.
but ok, i get that not every extension is perfect, so i'll add compatibility layer. fixed in dev, release later today.

**Comment by spreck** [comment]
Does this same issue apply to this plugin as well, https://github.com/Interpause/auto-sd-paint-ext. It's the Krita plugin.

I see "Status: Network error: Not Found", in Krita when I try to use. Also the tab is not showing up. 
![image](https://user-images.githubusercontent.com/69261029/235095013-12108ffc-33d2-4b8b-9d32-803135c148d1.png)

**Comment by vladmandic** [comment]
@spreck if tab is showing up in the ui, that means that extension loaded correctly. i don't know what is going on with krita -> extension communication as i don't use krita. best to report to extension author so he/she can check. if i need to make any changes to core repo to support extension, i'm willing to do that.

**Comment by spreck** [comment]
@vladmandic thanks for the response, I'll bring it up there. btw the tab is NOT showing up, the screenshot is what it is supposed to look like. It showed up in the 1111 fork but not since I switched to your branch

**Comment by vladmandic** [comment]
i've just tried installing extension and it shows up without problems after server restart.
other than that, i cannot test if its working or not as i don't have or use krita.

![image](https://user-images.githubusercontent.com/57876960/235211397-19ae4d38-40ab-4fef-a1b8-5291371b8fe8.png)

**Comment by Interpause** [comment]
What is the recommended way to check for the API functionality in your webUI? I do wish to maintain backwards compatibility.

A1111 adds API functionality on top of the FastAPI instance used by Gradio. Since I imagined some edge cases where the API might be undesirable, I decided to follow A1111's convention and only activate my API if A1111's API is active. Given in your webUI, the API is always active, is there some way for me to tell it is your webUI?
